trestle/transforms/utils/osco_helper.py

```diff
@@ -0,0 +1,444 @@
+                severity = lev1.get('severity')
+                weight = lev1.get('weight')
+                result = self._get_result(lev1)
+                rule_use = RuleUse(
+                    name,
+                    node,
+                    target,
+                    benchmark_href,
+                    benchmark_id,
+                    scanner_name,
+                    scanner_version,
+                    idref,
+                    version,
+                    time,
+                    result,
+                    severity,
+                    weight
+                )
+                yield rule_use
+
+    def rule_use_generator(self) -> RuleUse:
+        """Generate RuleUses by way of parsing the embedded XML."""
+        return self._parse_xml()
+
+
+class ResultsMgr():
+    """Represents collection of data to transformed into an AssessmentResult.results."""
+
+    default_timestamp = ResultsTransformer.get_timestamp()
+
+    def __init__(self, timestamp: t_timestamp = default_timestamp) -> None:
+        """Initialize."""
+        self.timestamp = timestamp
+        self.observation_list: t_observation_list = []
+        self.component_map: t_component_map = {}
+        self.findings_map: t_findings_map = {}
+        self.inventory_map: t_inventory_map = {}
+        self.results_map: t_results_map = {}
+        self.ns = 'http://ibm.github.io/compliance-trestle/schemas/oscal/ar/osco'
+
+    @property
+    def components(self) -> t_component_map:
+        """OSCAL components."""
+        return self.component_map
+
+    @property
+    def control_selections(self) -> List[t_control_selection]:
+        """OSCAL control selections."""
+        prop = []
+        prop.append(ControlSelection())
+        return prop
+
+    @property
+    def findings(self) -> List[t_finding]:
+        """OSCAL findings."""
+        control = ''
+        finding = Finding(uuid=str(uuid.uuid4()), title=control, description=control)
+        prop = [finding]
+        return prop
+
+    @property
+    def inventory(self) -> ValuesView[InventoryItem]:
+        """OSCAL inventory."""
+        return self.inventory_map.values()
+
+    @property
+    def local_definitions(self) -> t_local_definitions:
+        """OSCAL local definitions."""
+        prop = LocalDefinitions1()
+        prop.components = self.components
+        prop.inventory_items = list(self.inventory)
+        return prop
+
+    @property
+    def observations(self) -> List[t_observation]:
+        """OSCAL observations."""
+        return self.observation_list
+
+    @property
+    def reviewed_controls(self) -> t_reviewed_controls:
+        """OSCAL reviewed controls."""
+        prop = ReviewedControls(control_selections=self.control_selections)
+        return prop
+
+    @property
+    def result(self) -> t_result:
+        """OSCAL result."""
+        prop = Result(
+            uuid=str(uuid.uuid4()),
+            title='OpenShift',
+            description='Openshift',
```

Comment:
Do we want a description that is slightly more descriptive of how this content was produced?

Reply:
Fixed.